akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ReceiveBuilder.scala

```diff
@@ -61,31 +60,46 @@ class ReceiveBuilder[T] private (
    *
    * @param type type of message to match
    * @param handler action to apply when the type matches
-   * @return a new behavior with the specified handling appended
+   * @return this behavior builder
    */
-  def onMessageUnchecked[M <: T](`type`: Class[_ <: T], handler: Function[M, Behavior[T]]): ReceiveBuilder[T] =
-    withMessage(`type`, None, msg ⇒ handler.apply(msg.asInstanceOf[M]))
+  def onMessageUnchecked[M <: T](`type`: Class[_ <: T], handler: JFunction[M, Behavior[T]]): ReceiveBuilder[T] =
+    withMessage[M](OptionVal.Some(`type`.asInstanceOf[Class[M]]), OptionVal.None, handler)
 
   /**
    * Add a new case to the message handling matching equal messages.
    *
    * @param msg the message to compare to
    * @param handler action to apply when the message matches
-   * @return a new behavior with the specified handling appended
+   * @return this behavior builder
    */
   def onMessageEquals(msg: T, handler: Creator[Behavior[T]]): ReceiveBuilder[T] =
-    withMessage(msg.getClass, Some(_.equals(msg)), _ ⇒ handler.create())
+    withMessage(OptionVal.Some(msg.getClass), OptionVal.Some(new JPredicate[T] {
+      override def test(param: T): Boolean = param == (msg)
+    }), new JFunction[T, Behavior[T]] {
+      // invoke creator without the message
+      override def apply(param: T): Behavior[T] = handler.create()
+    })
+
+  /**
+   * Add a new case to the message handling matching any message. Subsequent `onMessage` clauses will
+   * never see any messages.
+   *
+   * @param handler action to apply for any message
+   * @return this behavior builder
+   */
+  def onAnyMessage(handler: JFunction[T, Behavior[T]]): ReceiveBuilder[T] =
```

Comment:
I think `. onAnyMessage` should `build()` automatically since it only makes sense to use as the last case. Then one can not do mistakes like
```
.onMessage(A.class, ...)
.onAnyMessage(...)
.onMessage(B.class, ...)
.build() 
```

That's how the command and event builders work.

Reply:
I considered doing that, but I don't think so, because we also have onSignal in here and a catchall messages before feels natural

Reply:
ah, good point